File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/selpersist/MarkTransientExample.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 
Imports System
Imports System.IO
Imports System.Collections
Imports com.db4o
Imports com.db4o.query

Namespace com.db4odoc.f1.selpersist
    Public Class MarkTransientExample
        Public Shared ReadOnly YapFileName As String = "formula1.yap"

        Public Shared Sub main(ByVal args() As String)
            ConfigureTransient()
            SaveObjects()
            RetrieveObjects()
        End Sub

        Public Shared Sub ConfigureTransient()
            Db4o.Configure().MarkTransient("com.db4odoc.f1.selpersist.FieldTransient")
        End Sub

        Public Shared Sub SaveObjects()
            File.Delete(YapFileName)
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim test As Test = New Test("Transient string", "Persistent string")
                oc.Set(test)
                Dim testc As TestCusomized = New TestCusomized("Transient string", "Persistent string")
                oc.Set(testc)
            Finally
                oc.Close()
            End Try
        End Sub

        Public Shared Sub RetrieveObjects()
            Dim oc As ObjectContainer = Db4o.OpenFile(YapFileName)
            Try
                Dim query As Query = oc.Query()
                query.Constrain(GetType(Object))
                Dim result As IList = query.Execute()
                ListResult(result)
            Finally
                oc.Close()
            End Try
        End Sub


        Public Shared Sub ListResult(ByVal result As IList)
            Console.WriteLine(result.Count)
            Dim x As Integer
            For x = 0 To result.Count - 1 Step x + 1
                Console.WriteLine(result(x))
            Next
        End Sub
    End Class
End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/selpersist/MarkTransientExample.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */
using System;
using System.IO;
using System.Collections;
using com.db4o;
using com.db4o.query;

namespace com.db4odoc.f1.selpersist
{
	public class MarkTransientExample
	{
		public readonly static string YapFileName = "formula1.yap";

		public static void Main(string[] args)
		{
			ConfigureTransient();
			SaveObjects();
			RetrieveObjects();
		}
		// end main

		public static void ConfigureTransient()
		{
			Db4o.Configure().MarkTransient("com.db4odoc.f1.selpersist.FieldTransient");
		}
		// end ConfigureTransient

		public static void SaveObjects()
		{
			File.Delete(YapFileName);
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				Test test = new Test("Transient string","Persistent string");
				oc.Set(test);
				TestCusomized testc = new TestCusomized("Transient string","Persistent string");
				oc.Set(testc);
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end SaveObjects

		public static void RetrieveObjects()
		{
			ObjectContainer oc = Db4o.OpenFile(YapFileName);
			try 
			{
				Query query = oc.Query();
				query.Constrain(typeof(object));
				IList result = query.Execute();
				ListResult(result);
			} 
			finally 
			{
				oc.Close();
			}
		}
		// end RetrieveObjects

		
		public static void ListResult(IList result)
		{
			Console.WriteLine(result.Count);
			for(int x = 0; x < result.Count; x++)
				Console.WriteLine(result[x]);
		}
		// end ListResult
	}
}
